pkg/auth/manager.go

```diff
@@ -29,6 +30,7 @@ type ipCache interface {
 type authHandler interface {
 	authenticate(*authRequest) (*authResponse, error)
 	authType() policy.AuthType
+	subscribeToRotatedIdentities() <-chan certs.CertificateRotationEvent
```

Comment:
tempted to say we should kind of "duplicate" the type `CertificateRotationEvent` within the authhandler - and convert at the boundary. but since we're also using other types it seems ok.

Reply:
we already have a shared struct structure here, i wouldn't like to do 1 to 1 conversion into types that do the same unless really necessary...